packages/jsdoc-plugins/lib/relation-fixer/getmissingdocletsdata.js

```diff
@@ -62,57 +68,60 @@ function getMissingDocletsData( docletCollection, childDoclet, options ) {
 	};
 }
 
-// Creates a <longname, doclet> map.
-//
-// @param {DocletCollection} doclets
-// @returns {Object}
-function createDocletMap( doclets ) {
-	const map = {};
-
-	for ( const doclet of doclets.getAll() ) {
-		map[ doclet.longname ] = doclet;
-	}
-
-	return map;
-}
-
-// Gets doclets from entities related to current doclet (e.g. implemented by it)
-// and matching criteria given in options.filter.
-function getDocletsToAdd( docletCollection, childDoclet, options = {} ) {
+/**
+ * Gets doclets from entities related to current doclet (e.g. implemented by it)
+ * and matching criteria given in options.filter.
+ *
+ * @param {DocletCollection} docletCollection
+ * @param {Doclet} childDoclet
+ * @param {Options} options
+ */
+function getDocletsToAdd( docletCollection, childDoclet, options ) {
 	if ( !isNonEmptyArray( childDoclet[ options.relation ] ) ) {
 		return [];
 	}
 
 	// Longnames of doclets which are related ( extended, mixed, implemented ) to childDoclet.
 	const ancestors = childDoclet[ options.relation ];
 
-	return ancestors.reduce( ( docletsToAdd, longname ) => {
-		const toAdd = docletCollection.get( `memberof:${ longname }` ).filter( d => {
-			let isMatchingFilterOptions = true;
+	const docletToAdd = ancestors.reduce( ( /** @type {Array.<Doclet>} */ docletsToAdd, longname ) => {
+		const toAdd = docletCollection.get( `memberof:${ longname }` ).filter( doclet => {
+			let matchingFilterOptions = true;
```

Comment:
This variable can be removed. Following codes could be replaced:

* `matchingFilterOptions = false` => `return false`
* `return matchingFilterOptions`=> `return true`